sympy/functions/combinatorial/numbers.py

```diff
@@ -841,6 +841,116 @@ def _eval_evalf(self, prec):
         if self.args[0].is_number:
             return self.rewrite(C.gamma)._eval_evalf(prec)
 
+
+#----------------------------------------------------------------------------#
+#                                                                            #
+#                           Genocchi numbers                                 #
+#                                                                            #
+#----------------------------------------------------------------------------#
+
+
+class genocchi(Function):
+    r"""
+    Genocchi numbers
+
+    The Genocchi numbers are a sequence of integers G_n that satisfy the
+    relation::
+
+                           oo
+                         ____
+                         \   `
+                 2*t      \         n
+                ------ =   \   G_n*t
+                 t         /   ------
+                e  + 1    /      n!
+                         /___,
+                         n = 1
+
+    Examples
+    ========
+
+    >>> from sympy import Symbol
+    >>> from sympy.functions import genocchi
+    >>> [genocchi(n) for n in range(8)]
+    [1, -1, 0, 1, 0, -3, 0, 17]
```

Comment:
I think here even 0 is included and it raises ValueError because its non-positive